### PR TITLE
Date processor logstash converter

### PR DESCRIPTION
**Pull request by asifsmohammed**
### Description

- Logstash converter for date processor with custom mapper for match attribute
- Integration test for converter

 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/DateLogstashPluginAttributesMapper.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @SuppressWarnings("unchecked")
+    private Map<String, List<String>> convertMatchAttribute(LogstashAttribute matchAttribute) {
+        Map<String, List<String>> dataPrepperDateMatch = new HashMap<>();
+
+        final LogstashValueType logstashDateMatchValueType = matchAttribute.getAttributeValue().getAttributeValueType();
+        final Object logstashDateMatchValue = matchAttribute.getAttributeValue().getValue();
+
+        if (logstashDateMatchValueType.equals(LogstashValueType.ARRAY)) {
+            final List<String> logstashDateMatchCastValue = (List<String>) logstashDateMatchValue;
+            if (logstashDateMatchCastValue.size() >= 2) {
```
Comment (cmanning09) [inline data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/DateLogstashPluginAttributesMapper.java:55]: What will happen if this is not true? Will we throw an exception when running Data Prepper for an invalid date processor config? Should we fail sooner?

File: data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/DateLogstashPluginAttributesMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.opensearch.dataprepper.logstash.mapping.DateLogstashPluginAttributesMapper.LOGSTASH_DATE_MATCH_ATTRIBUTE_NAME;
+import static org.opensearch.dataprepper.logstash.model.LogstashValueType.ARRAY;
+
+class DateLogstashPluginAttributesMapperTest {
```
Comment (cmanning09) [inline data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/DateLogstashPluginAttributesMapperTest.java:28]: Let's add some test cases to test the edge cases I called out:
- missing match
- invalid match list size

File: data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/DateLogstashPluginAttributesMapper.java (hunk 1 above)
Comment (asifsmohammed) [reply data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/DateLogstashPluginAttributesMapper.java:55]: I think we should do the validation while running Data Prepper and let the conversion generate an YAML file.

**Review by cmanning09 — APPROVED** [review]

**Review by graytaylor0 — APPROVED** [review]
